include/directx/d3dx12_pipeline_state_stream.h

```diff
@@ -1362,126 +1362,197 @@ inline HRESULT D3DX12ParsePipelineStream(const D3D12_PIPELINE_STATE_STREAM_DESC&
             pCallbacks->ErrorDuplicateSubobject(SubobjectType);
             return E_INVALIDARG; // disallow subobject duplicates in a stream
         }
+        switch (SubobjectType)
```

Comment:
since all these sizes are static, this seems like a switch statement isn't particular efficient as an implementation.

Why note create a ``const static`` array indexed by SubobjectType that has the `sizeof` in it? That way the look-up in trivial and the information is encoded in the readonly dataseg instead of as a code jumptable? The only reason the size was in the original switch statement is it already needed the code-paths for the callbacks.

Some use of ``static_assert`` should make it easy to notice when new pipeline state values are added as well as the runtime error.

Reply:
Initially, I tried an array approach, but I thought it was not safe enough